src/borg/compress.pyx

```diff
@@ -1,9 +1,41 @@
+"""
+borg.compress
+=============
+
+Compression is applied to chunks after ID hashing (so the ID is a direct function of the
+plain chunk, compression is irrelevant to it), and of course before encryption.
+
+Borg has a flexible scheme for deciding which compression to use for chunks.
+
+First, there is a global default set by the --compression command line option,
+which sets the .compressor attribute on the Key.
+
+For chunks that emanate from files CompressionDecider may set a specific
+Compressor based on patterns (this is the --compression-from option). This is stored
+as a Compressor instance in the "compressor" key in the Chunk's meta dictionary.
```

Comment:
considering we might soon send this over some socket, I'ld rather like this to be a simple object (specification) than a life Compressor class instance.